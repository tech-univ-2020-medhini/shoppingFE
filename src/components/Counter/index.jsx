import React, { useState } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import './index.css';


function Counter(props) {
  const {
    id, setCartCount, cartCount, max,
  } = props;
  const { cart } = props;
  const [cartValue, setCartValue] = useState(cart);
  const onClickCounter = (direc) => {
    console.log(max);
    if (direc === '+' && cartValue < max) {
      const patchCart = async () => {
        axios.patch(`http://localhost:8080/products/${id}/cart`, {
          value: cart + 1,
        });
        setCartValue(cartValue + 1);
        setCartCount(cartCount + 1);
      };
      patchCart();
    } else if (direc === '-' && cartValue > 0) {
      const patchCart = async () => {
        axios.patch(`http://localhost:8080/products/${id}/cart`, {
          value: cart - 1,
        });
        setCartValue(cartValue - 1);
        setCartCount(cartCount - 1);
      };
      patchCart();
    }
  };
  return (
    <div className="counter">
      <button className="counter-button" onClick={() => onClickCounter('-')} type="button">-</button>
      {cartValue}
      <button className="counter-button" onClick={() => onClickCounter('+')} type="button">+</button>
    </div>

  );
}
Counter.propTypes = {
  id: PropTypes.number.isRequired,
  cart: PropTypes.number.isRequired,
  cartCount: PropTypes.number.isRequired,
  max: PropTypes.number.isRequired,
  setCartCount: PropTypes.func.isRequired,
};

export default Counter;
