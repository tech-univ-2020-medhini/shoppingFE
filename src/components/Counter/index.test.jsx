import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Counter from './index';


describe('the counter component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Counter />);

    expect(asFragment()).toMatchSnapshot();
  });
  it('should increment its value when increment is pressed', () => {
    const { getByTestId } = render(<Counter />);

    fireEvent.click(getByTestId('test-increment'));
    expect(getByTestId('counter-div')).toHaveTextContent('-+');
  });
});
