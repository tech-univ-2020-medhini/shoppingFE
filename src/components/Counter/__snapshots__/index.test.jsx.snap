// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the counter component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="counter"
    data-testid="counter-div"
  >
    <button
      class="counter-button"
      data-testid="test-decrement"
      type="button"
    >
      -
    </button>
    <button
      class="counter-button"
      data-testid="test-increment"
      type="button"
    >
      +
    </button>
  </div>
</DocumentFragment>
`;
